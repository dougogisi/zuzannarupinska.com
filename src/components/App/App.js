import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export const App = props => <h1>{props.name}</h1>;

App.propTypes = {
  name: PropTypes.string.isRequired,
};

const mapStateToProps = state => {
  return {
    name: state.name
  };
};

export default connect(mapStateToProps)(App);
