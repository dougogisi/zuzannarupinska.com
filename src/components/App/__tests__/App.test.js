import React from 'react';
import { mount } from 'enzyme';
import { App } from '../App';

describe('<App />', () => {

  test('Should render correctly', () => {
    const wrapper = mount(<App yo="Test :-)" />);

    expect(wrapper).toMatchSnapshot();
  });

});
