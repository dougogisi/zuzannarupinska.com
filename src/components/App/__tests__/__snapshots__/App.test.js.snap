// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should render correctly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <App
    yo="Test :-)"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "yo": "Test :-)",
    },
    "ref": null,
    "rendered": Object {
      "instance": <h1>
        Test :-)
      </h1>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Test :-)",
      },
      "ref": null,
      "rendered": Array [
        "Test :-)",
      ],
      "type": "h1",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "yo": "Test :-)",
      },
      "ref": null,
      "rendered": Object {
        "instance": <h1>
          Test :-)
        </h1>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test :-)",
        },
        "ref": null,
        "rendered": Array [
          "Test :-)",
        ],
        "type": "h1",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
